### PR TITLE
Upgraded the KCL dependency to minimum 1.7.1

### PR DESCRIPTION
**Pull request by jdoc**
[Related Issue](https://github.com/awslabs/dynamodb-streams-kinesis-adapter/issues/6)

### CODE REVIEW

**Comment by joelittlejohn** [comment]
Would be great to see this one merged. I'd really like to be able to use a new KCL (1.7.4) that includes https://github.com/awslabs/amazon-kinesis-client/pull/135 as it's a fairly important fix. I'm seeing a huge number of threads constantly cycled whilst using this library.

Seems like this project could do with some attention as it's falling behind a little but is still the best and recommended way to interact with DynamoDB Streams from Java (AFAIK).

**Comment by joelittlejohn** [comment]
@pfifer Sorry to bother you directly if you're not involved, but do you know if this project currently has maintainer?

File: src/main/java/com/amazonaws/services/dynamodbv2/streamsadapter/StreamsMultiLangDaemon.java [hunk 1]
```diff
@@ -58,7 +59,7 @@ public static void main(String[] args) {
         ExecutorService executorService = config.getExecutorService();
 
         Worker worker = new StreamsWorker(
-                config.getRecordProcessorFactory(),
+                (IRecordProcessorFactory) config.getRecordProcessorFactory(),
```
Comment (tkbalajiprasad) [inline src/main/java/com/amazonaws/services/dynamodbv2/streamsadapter/StreamsMultiLangDaemon.java]: Can you please explain what is the motivation behind this explicit casting? It already used to be of type 'IRecordProcessorFactory' correct?

**Comment by amcp** [comment]
We released [support for KCL 1.7.5](https://github.com/awslabs/dynamodb-streams-kinesis-adapter/tree/1.2.0) today. Closing.

**Pull request closed without merging**